internal/services/managedapplications/managed_application_definition_resource.go

```diff
@@ -193,105 +190,99 @@ func resourceManagedApplicationDefinitionRead(d *pluginsdk.ResourceData, meta in
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationDefinitionID(d.Id())
+	id, err := applicationdefinitions.ParseApplicationDefinitionID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			log.Printf("[INFO] Managed Application Definition %q does not exist - removing from state", d.Id())
```

Comment:
_minor_ but for consistency we can make this:

```suggestion
			log.Printf("[INFO] %s does not exist - removing from state", *id)
```